### PR TITLE
Update haz3l-tests to OCaml 4.14.0

### PR DESCRIPTION
**Pull request by dedbox**
Lots of minor changes needed for compiling under OCaml 4.14. It makes the LSP server very happy.

### CODE REVIEW

**Comment by dm0n3y** [comment]
we'd have to update our release builds too, not sure how annoying that would be. also what's with the `core` to `core3` rename?

**Comment by dedbox** [comment]
> what's with the `core` to `core3` rename?

The latest version of the Jane St. package `core_kernel` renames the module `Core_kernel` to `Core`. This is the only version supported by OCaml 4.14.

**Comment by dm0n3y** [comment]
Instead of renaming our core, would it be possible to make only our `Util` library depend on jane street core and expose it via the `Util` library elsewhere as `Jane` or something?

File: docs/quick-and-dirty-ocaml414.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+These instructions assume you want a local opam switch.
```
Comment (cyrus-) [inline docs/quick-and-dirty-ocaml414.md]: Can you make this into a shell script?

**Review by cyrus- — CHANGES_REQUESTED** [review]

File: docs/quick-and-dirty-ocaml414.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+``` shell
+eval $(opam env --switch=. --set-switch)
+```
+
+3. From somewhere outside the Hazel source tree, clone the following opam package repos:
+
+``` shell
+cd ..
+git clone 'git@github.com:janestreet/async_js.git'
+git clone 'git+https://github.com/janestreet/incr_dom.git'
```
Comment (cyrus-) [inline docs/quick-and-dirty-ocaml414.md]: git+https doesn't parse on my machine

Comment (dedbox) [reply docs/quick-and-dirty-ocaml414.md]: > git+https doesn't parse on my machine

Not sure what you mean. What did you try? I gave `opam pin` some https URLs and it converted them into what you see there.

**Review by cyrus- — APPROVED** [review]

**Pull request merged**